### ISSUE TITLE
YIR Landing Page: Fix domain to blog identification

### ISSUE THREAD

**Issue by markbiek** [Automattic/wp-calypso#85226]
Full domain to blog identification is not in place yet, so we're not correctly finding the blog ID from mapped domains (e.g. distributed.blog)

**Comment by daledupreez** [comment]
We should be able to handle this via logic pretty similar to the following snippet, though the specifics here are based on the code from D131860-code:
```php
			$year = (int) $matches[2];
			if ( str_ends_with( $matches[1], '.wordpress.com' ) ) {
				$bid = get_blog_id_from_url( $matches[1] );
			} else {
				$bid = (int) get_blog_id_from_domain( $matches[1] );
			}
```

**Comment by daledupreez** [comment]
Proposed fix in D131942-code

**Comment by daledupreez** [comment]
The fix was shipped yesterday in D131942-code.
